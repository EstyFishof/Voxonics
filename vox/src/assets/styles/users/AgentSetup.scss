.wrap-agent-setup {
  padding-bottom: 36px;

  .line-beetwen-sections {
    max-width: 632px;
    margin-left: 48px;
  }

  .wrap-head {
    margin-bottom: 16px;
  }

  .agent-setup {

    &__first-part,
    &__second-part {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      align-items: baseline;
      margin-left: 48px;
      margin-bottom: 36px;

      .hr {
        max-width: 632px;
        margin-left: 0;
      }
    }

    &__second-part {

      .item-container {
        position: relative;
        margin-top: 24px;

        .description {
          margin-left: 10px;
        }

        &:nth-child(1) {
          margin-right: 48px;
        }

        &:nth-child(2) {
          margin-right: 48px;
        }

        &:nth-child(3) {
          margin-right: 48px;
        }
      }


      .options-label {
        font-size: 18px;
        font-weight: 700;
        text-align: left;

        .input-field {
          background: var(--input-bg);
          width: 292px;
          height: 35px;
          margin-top: 12px;
          padding: 5px 12px;
        }

        .triangle {
          width: calc(104% - 7px);
          height: 35px;

          img {
            margin-right: 10px;
          }
        }

        .list-related-numbers ::v-deep {
          overflow: auto;
          margin-top: 1px;

          li {
            color: var(--white);
            height: 44px;
            padding-left: 16px;
            font-weight: 300;
            text-transform: capitalize;
          }
        }
      }

      .calls-info {
        position: absolute;
        margin-top: 16px;
        font-size: 16px;
        font-weight: 300;
        white-space: nowrap;

        span {
          color: #515960;
        }
      }
    }
  }
}

.agent-setup__second-part .item-container label {

  ul {
    position: absolute;
    z-index: 999;
    max-height: 150px;

    li {
      background-color: var(--paginator-item);
      color: var(--font-panel);
      padding: 5px 0 5px 7px;

      &:hover {
        background-color: var(--light-blue);
        color: white;
      }
    }
  }
}

.collapse {
  height: 16px;
  position: sticky;
  top: 0;
  background-color: var(--dropdown-list-collapse-bg);
  border-radius: 4px;
  margin-bottom: -2px;
  z-index: 999;
  box-shadow: 0 5px 16px rgba(0, 0, 0, 0.5);

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;

  cursor: pointer;
  font-weight: 300 !important;
  font-size: 12px !important;
  line-height: 14px !important;

  .text {
    font-weight: 300;
    font-size: 12px;
    line-height: 14px;
    color: var(--dropdown-list-collapse-text-color);
  }

  .amount {
    margin-right: 24px;
    color: var(--dropdown-list-collapse-text-color);
  }
}
