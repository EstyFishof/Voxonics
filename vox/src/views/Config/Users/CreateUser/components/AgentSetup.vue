<template>
  <div class="wrap-agent-setup">

    <div class="wrap-head">
      <h2 class="wrap-head-title">AGENT SETUP</h2>
      <svg @click="showOptions = !showOptions;chooseDepartments({})" class="arrow" :class="{'down': showOptions}"
           viewBox="0 0 10 18"
           fill="transparent"
           xmlns="http://www.w3.org/2000/svg">
        <path d="M9 1L1 9L9 17" stroke-width="2" stroke="var(--light-blue)" stroke-linecap="round"
              stroke-linejoin="round"/>
      </svg>
      <span class="hr"></span>
    </div>

    <div v-show="showOptions" class="wrap-body" @click="hideListOptions(); indexRef='false'" v-click-outside="hide"
         ref="hideList">

      <div class="agent-setup__second-part">
        <div class="item-container" :ref="'Role'" @click="indexRef = 'Role'">
          <label class="options-label">
            <span class="options-title">Role</span>
            <input @click.stop="hideListRole" class="input-field" :value="role" type="text" readonly/>
            <span class="triangle" :class="{'rotate-arrow': showRole}">
              <img src="@/assets/img/icons/arrow-down-white.svg" alt="select">
            </span>
            <DropdownList v-show="showRole" :classNameList="'list-related-numbers agent-setup-user-list'" :dataList="listOfRoles"
                @clickFunc="(item) => selectRole(item)" :itemText="(item) => item" />
          </label>
        </div>

        <div class="item-container">
          <label class="options-label" :ref="'Departments Settings'" @click="indexRef = 'Departments Settings'">
            <span class="options-title">Departments Settings</span>
            <input 
              class="input-field dep" 
              type="text" 
              :value="inputDepValue" 
              @input="searchDeps($event.target.value)"
              maxlength="16"
            />
            <span @click="toggleVisible(!show, 'show')" class="triangle" :class="{'rotate-arrow': show}">
              <img src="@/assets/img/icons/arrow-down-white.svg" alt="select">
            </span>

            <DropdownList v-show="show" :classNameList="'list-related-numbers'" :dataList="getDep"
                @clickFunc="(item) => {departmentsToArray(item); chooseDepartments(item)}" :itemText="(item) => item.name"
                :clickFuncAll="null"  @scrollFunc="(event) => funScroll(event)" :showItem="(item) => topList && (topList.filter(dep => dep.uid === item.uid)).length ? !collapse : true"
                :classNameItemComputed="(item) => (topList.filter(dep => dep.uid === item.uid)).length ? collapse ? 'selected collapsed' : 'selected' : ''">
              <span class="collapse" @click="collapse = !collapse">
                <img class="button" :src="collapse ? extendImg : collapseImg" alt="collapse">
                <span class="text">Selected users available to you:</span>
                <span class="amount">{{ topList.length }}</span>
              </span>
            </DropdownList>

          </label>
        </div>
      </div>
      <span class="hr line-beetwen-sections"></span>
      <div class="agent-setup__second-part">
        <div class="item-container">
          <label class="options-label">
            <span class="options-title">Calls Direction*</span>
            <span class="input-field">
              00 - all
            </span>
          </label>
          <p class="calls-info">
            <span>*You can change it by contacting us</span>
          </p>
        </div>

        <div class="item-container">
          <label class="options-label">
            <span class="options-title">V-number**</span>
            <input class="input-field num" type="text" value="Preferred" readonly
                   style="pointer-events: none; outline: none">
          </label>
          <p class="calls-info">
            <span>**You can change it in Outbound routing</span>
          </p>
        </div>

      </div>
    </div>
  </div>
</template>

<script>

import { mapGetters } from "vuex"
import ClickOutside from 'vue-click-outside'
import collapseImg from '@/assets/img/icons/collapse.svg'
import extendImg from '@/assets/img/icons/extend.svg'
import DropdownList from '@/components/DropdownList'
// import {DepartmentGettersApi} from '@/API/getters'
import parseCountries from '@/lib/selectedCountries'
import { clickCheck } from '@/lib/clickCheck.js'
import _ from 'lodash'

export default {
  name: "AgentSetup",
  components: {
    DropdownList
  },
  data() {
    return {
      unsubscribeStore: () => {
      },
      collapseImg,
      extendImg,
      collapse: false,
      show: false,
      showOptions: true,
      search: "",
      showSearchResults: false,
      markedDepartments: "0 selected",
      selectedDepartments: [],
      role: "agent",
      showRole: false,
      perPage: 9,
      count: 10,
      timer: null,
      topList: [],
      bottomList: [],
      getDep: [],
      indexRef: '',
      depSearchValue: '',
    }
  },
  computed: {
    ...mapGetters(["getGlobalDepartments", "getSelectedDepartments", "VNumber", 'getPaginatorDepartment']),
    listOfRoles() {
      return this.$store.state.displayConfig.displayConfig.createUser.agentSetup.listRoles
    },
    inputDepValue() {
      if (this.show) {
        return this.depSearchValue
      } else {
        return this.markedDepartments
      }
    }
  },
  methods: {
    departmentsToArray(item) {
      const obj = parseCountries.changeCountry(item, JSON.parse(JSON.stringify(this.topList)), JSON.parse(JSON.stringify(this.bottomList)), 'uid')
      this.topList = obj.topList
      this.bottomList = obj.bottomList
      this.getDep = obj.getCountries
    },
    searchDeps(value) {
      this.depSearchValue = value
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch(), 800)
    },
    funScroll({target}) {
      if (target.clientHeight + target.scrollTop === target.scrollHeight) {
        if (this.getPaginatorDepartment.paginator.count > this.perPage) {
          this.perPage += 9
          this.fetch()
        }
      }
    },
    fetch() {
      this.$store.dispatch('setGlobalDepartmentsList', {peerPage: this.perPage, search: this.depSearchValue})
    },
    hideListOptions() {
      clickCheck(this, this.indexRef, this.$refs)
    },
    hide() {
      this.toggleVisible(false, 'show')
      this.showRole = false
    },
    hideListRole() {
      this.showRole = !this.showRole
    },
    selectRole(role) {
      this.role = role
      this.$store.dispatch('setRole', this.role)
    },
    toggleVisible(bool, param) {
      this[param] = bool
    },
    chooseDepartments(department) {
      if (department) {
        this.defaultDepartment()
      }
      let deps = ""
      if (!this.topList.length) {
        this.markedDepartments = "0 selected"
      } else {
        this.topList.forEach(item => deps += item.name + ", ")
        this.markedDepartments = `${this.topList.length} selected (${deps})`
        this.markedDepartments = this.markedDepartments.slice(0, 30) + '...)'
      }
    },
    defaultDepartment() {
      this.$store.dispatch('departmentsSelected', this.topList)
    },
    initialisationList() {
      this.topList = JSON.parse(JSON.stringify(this.getSelectedDepartments))
      const allList = _.uniqBy(_.concat(this.getDep, this.getGlobalDepartments), d => {
        return d.uid
      })
      this.bottomList = allList.filter(dep => !this.topList.some(item => dep.uid === item.uid))
      this.getDep = _.concat(this.topList, this.bottomList)
    }
  },
  created() {
    this.$store.dispatch('setRole', this.role)
  },
  mounted() {
    this.popupItem = this.$refs.hideList
    this.chooseDepartments()
    this.initialisationList()
    this.unsubscribeStore = this.$store.subscribe(mutation => {
      if (mutation.type === 'pushRelatedDepartments' || mutation.type === 'pushSelectedDepartments') {
        this.initialisationList()
        this.chooseDepartments()
      }
    })
  },
  beforeDestroy() {
    this.unsubscribeStore()
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/users/AgentSetup.scss"></style>
<!-- <style lang="scss">
  .list-related-numbers.agent-setup-user-list {
    overflow: auto;
    margin-top: 1px;

    li {
      color: var(--white);
      height: 44px;
      padding-left: 16px;
      font-weight: 300;
      text-transform: capitalize;
    }
  }
</style> -->
