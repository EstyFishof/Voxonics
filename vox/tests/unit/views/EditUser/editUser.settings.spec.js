import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import AgentSetup from '@/views/Config/Users/EditUser/components/AgentSetup';
import QSettings from '@/views/Config/Users/EditUser/components/QSettings';
import FileLoader from '@/views/Config/Users/EditUser/components/FileLoader';
import user from "./__mock__/user";
import VueTippy, { TippyComponent } from "vue-tippy";


const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);


describe('agent setup', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters, actions, store;
    beforeEach(() => {
        getters = {
            userInfoVuex: () => user,
            getSearchUsers: () => [],
            getRelatedDepartments: () => [],
            departments: () => [{name: '', uid: 123}],
            getOutboundDirection: () => '123',
            outInfo: () => ([{ uid: 123, number: 123 }]),
            VNumber: () => '123',
            userRole: () => 'manager',
            userInfo: () => {123;}
        };
        actions = {
            changeRole: jest.fn(),
            changeDeps: jest.fn(),
            changeVNumber: jest.fn()
        };
        store = new Vuex.Store({ actions, getters });
    });
    it('it should  Vue instance', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        expect(wrapper.vm).toBeTruthy();
    });
    // it('Check snapshot', () => {
    //     const wrapper = shallowMount(AgentSetup, { localVue, store });
    //     expect(wrapper.element).toMatchSnapshot();
    // })
    it('markedDepartments  defined', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        expect(wrapper.vm.markedDepartments).toBeDefined();
    });
    it('selectRole', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'selectRole');
        wrapper.vm.selectRole({ target: 'classname' });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('toggle visible password', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'toggleVisible');
        wrapper.vm.toggleVisible();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('search user empty', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'searchUser');
        wrapper.vm.searchUser();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('search user edit', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        wrapper.setData({ search: 'aaa' });
        const spy = jest.spyOn(wrapper.vm, 'searchUser');
        wrapper.vm.searchUser();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('choose agent edit', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        wrapper.vm.chooseAgent();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('choose departments empty args', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'chooseDepartments');
        wrapper.vm.chooseDepartments();
        expect(spy).toHaveBeenCalled();
        let deps = "";
        let markedDepartments = `${wrapper.vm.departments.length} selected (${deps})`;
        expect(markedDepartments).toBe('1 selected ()');
        markedDepartments = markedDepartments.slice(0, 30) + '...';
        expect(markedDepartments).toBe('1 selected ()...');
        spy.mockRestore();
    });
    it('choose departments', () => {
        const wrapper = shallowMount(AgentSetup, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'chooseDepartments');
        wrapper.vm.chooseDepartments({ name: '', uid: 123 });
        expect(spy).toHaveBeenCalled();
        let deps = "";
        let markedDepartments = `${wrapper.vm.departments.length} selected (${deps})`;
        expect(markedDepartments).toBe('1 selected ()');
        markedDepartments = markedDepartments.slice(0, 30) + '...';
        expect(markedDepartments).toBe('1 selected ()...');
        spy.mockRestore();
    });
    it('chooseNumber', () => {
        const mock = jest.fn();
        const list = {
            render() {},
            methods: {
                mock
            }
        };
        const wrapper = shallowMount(AgentSetup, { localVue, store, stubs: {list} });
        // wrapper.vm.$refs.list.mock()
        const spy = jest.spyOn(wrapper.vm, 'chooseNumber');
        wrapper.vm.chooseNumber({ number: 38096 });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('choose VNumber', () => {
        getters = {
            getSearchUsers: () => [],
            getRelatedDepartments: () => [],
            departments: () => [{name: ''}],
            getOutboundDirection: () => '123',
            outInfo: () => [{ uid: 123, number: 123 }],
            VNumber: () => 123,
            userRole: () => 'agent',
            userInfoVuex: () => user,
        };
        actions = {
            changeRole: jest.fn(),
            changeDeps: jest.fn(),
            changeVNumber: jest.fn()
        };
        store = new Vuex.Store({ actions, getters });
        // const wrapper = shallowMount(AgentSetup, { localVue, store });
        const mock = jest.fn();
        const list = {
            render() {},
            methods: {
                mock
            }
        };
        const wrapper = shallowMount(AgentSetup, { localVue, store, stubs: {list} });
        // wrapper.vm.$refs.list.mock()
        const spy = jest.spyOn(wrapper.vm, 'chooseNumber');
        wrapper.vm.chooseNumber({ uid: 123, number: 123 });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('result', () => {
        const mock = jest.fn();
        const list = () => ({
            render: () => {
                return this.$refs;
            },
        });
        const wrapper = shallowMount(AgentSetup, { localVue, store, mocks: {$refs: list} });
        // wrapper.vm.$refs.list.mock()
        const result = wrapper.vm.outInfo.map(num => {
            if (num.uid === wrapper.vm.VNumber) {
                wrapper.vm.selectedNumber = num.number;
                return num;
            }
        });
        expect(result).toHaveLength(1);
    });

});

describe('queue settings / edit', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    const reactiveData = [
        'showOptions',
        'relatedQueues',
        'selectedQueues',
        'dataQueues',
        'viewQueues',];
    let getters, actions, store;
    beforeEach(() => {
        getters = {
            getRelatedQueues: () => [],
            userQueues: () => [],
        };
        actions = {
            usersQueuesSelected: jest.fn(),
        };
        store = new Vuex.Store({ actions, getters });
    });
    it('should Vue instance', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot qs', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('vuex data defined', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        expect(wrapper.vm.getRelatedQueues).toBeDefined();
        expect(wrapper.vm.userQueues).toBeDefined();
    });
    test.each(reactiveData)('data defined', (property) => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        expect(wrapper.vm[property]).toBeDefined();
    });
    it('selectQ empty', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'selectQ');
        wrapper.vm.selectQ({ uid: '123' });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectQ', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'selectQ');
        wrapper.setData({ selectedQueues: [{ uid: '123' }] });
        wrapper.vm.selectQ({ uid: '123' });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteQ empty', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'deleteQ');
        wrapper.vm.deleteQ();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteQ', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'deleteQ');
        wrapper.vm.deleteQ({ uid: '123' });
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.selectedQueues).toStrictEqual([]);
        spy.mockRestore();
    });
    it('select all Queues', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'selectAllQ');
        wrapper.vm.selectAllQ();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('remove all Queues', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'removeAllQ');
        wrapper.vm.removeAllQ();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('dispatch selected Queues', () => {
        const wrapper = shallowMount(QSettings, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'dispatchSelectedQueues');
        wrapper.vm.dispatchSelectedQueues(jest.fn(), '');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});

describe('FileLoader', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters;
    let store;
    beforeEach(() => {
        getters = { userUid: () => '', userInfoVuex: () => '', userAva: () => ''};
        store = new Vuex.Store({ getters });
    });
    it('should  Vue instance', () => {
        const wrapper = shallowMount(FileLoader, { localVue, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        const wrapper = shallowMount(FileLoader, { localVue, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('handleFileUpload', () => {
        const wrapper = shallowMount(FileLoader, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'handleFileUpload');
        wrapper.vm.handleFileUpload();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});
