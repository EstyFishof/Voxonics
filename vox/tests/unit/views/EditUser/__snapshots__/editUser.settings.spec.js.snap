// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileLoader Check snapshot 1`] = `
<div
  class="container"
>
  <div
    class="personal-img"
  >
    <svg
      class="personal-img__user-img"
      fill="none"
      height="90"
      viewBox="0 0 68 90"
      width="68"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="inside"
        clip-rule="evenodd"
        d="M66 83.2228C66 85.8601 63.9566 88.0005 61.4286 88.0005C58.9006 88.0005 56.8571 85.8601 56.8571 83.2228C56.8571 70.0505 46.6034 59.3339 34 59.3339C21.3966 59.3339 11.1429 70.0505 11.1429 83.2228C11.1429 85.8601 9.09943 88.0005 6.57143 88.0005C4.04343 88.0005 2 85.8601 2 83.2228C2 64.7806 16.3589 49.7784 34 49.7784C51.6411 49.7784 66 64.7806 66 83.2228M34 11.5584C39.0423 11.5584 43.1429 15.8441 43.1429 21.114C43.1429 26.3838 39.0423 30.6695 34 30.6695C28.9578 30.6695 24.8572 26.3838 24.8572 21.114C24.8572 15.8441 28.9578 11.5584 34 11.5584M34.0001 40.2221C44.0847 40.2221 52.2858 31.6508 52.2858 21.111C52.2858 10.5713 44.0847 2 34.0001 2C23.9156 2 15.7144 10.5713 15.7144 21.111C15.7144 31.6508 23.9156 40.2221 34.0001 40.2221"
        fill-rule="evenodd"
      />
       
      <path
        class="out"
        d="M64 83.2228C64 84.841 62.7684 86.0005 61.4286 86.0005V90.0005C65.1447 90.0005 68 86.8792 68 83.2228H64ZM61.4286 86.0005C60.0887 86.0005 58.8571 84.841 58.8571 83.2228H54.8571C54.8571 86.8792 57.7124 90.0005 61.4286 90.0005V86.0005ZM58.8571 83.2228C58.8571 69.03 47.7903 57.3339 34 57.3339V61.3339C45.4166 61.3339 54.8571 71.0709 54.8571 83.2228H58.8571ZM34 57.3339C20.2097 57.3339 9.14286 69.03 9.14286 83.2228H13.1429C13.1429 71.0709 22.5834 61.3339 34 61.3339V57.3339ZM9.14286 83.2228C9.14286 84.841 7.91129 86.0005 6.57143 86.0005V90.0005C10.2876 90.0005 13.1429 86.8792 13.1429 83.2228H9.14286ZM6.57143 86.0005C5.23156 86.0005 4 84.841 4 83.2228H0C0 86.8792 2.85529 90.0005 6.57143 90.0005V86.0005ZM4 83.2228C4 65.8012 17.5455 51.7784 34 51.7784V47.7784C15.1722 47.7784 0 63.76 0 83.2228H4ZM34 51.7784C50.4545 51.7784 64 65.8012 64 83.2228H68C68 63.76 52.8278 47.7784 34 47.7784V51.7784ZM34 13.5584C37.8555 13.5584 41.1429 16.8645 41.1429 21.114H45.1429C45.1429 14.8237 40.2292 9.55844 34 9.55844V13.5584ZM41.1429 21.114C41.1429 25.3634 37.8555 28.6695 34 28.6695V32.6695C40.2292 32.6695 45.1429 27.4043 45.1429 21.114H41.1429ZM34 28.6695C30.1446 28.6695 26.8572 25.3634 26.8572 21.114H22.8572C22.8572 27.4043 27.7709 32.6695 34 32.6695V28.6695ZM26.8572 21.114C26.8572 16.8645 30.1446 13.5584 34 13.5584V9.55844C27.7709 9.55844 22.8572 14.8237 22.8572 21.114H26.8572ZM34.0001 42.2221C45.2716 42.2221 54.2858 32.6712 54.2858 21.111H50.2858C50.2858 30.6303 42.8978 38.2221 34.0001 38.2221V42.2221ZM54.2858 21.111C54.2858 9.55086 45.2716 0 34.0001 0V4C42.8978 4 50.2858 11.5917 50.2858 21.111H54.2858ZM34.0001 0C22.7287 0 13.7144 9.55086 13.7144 21.111H17.7144C17.7144 11.5917 25.1024 4 34.0001 4V0ZM13.7144 21.111C13.7144 32.6712 22.7287 42.2221 34.0001 42.2221V38.2221C25.1024 38.2221 17.7144 30.6303 17.7144 21.111H13.7144Z"
      />
    </svg>
     
    <label>
      <span>
        Upload your photo
      </span>
       
      <input
        id="file"
        type="file"
      />
    </label>
     
    <button>
      <img
        alt="For submit press"
        src="@/assets/img/icons/upload-img-plus.svg"
      />
    </button>
     
    <h1>
      
    </h1>
  </div>
</div>
`;

exports[`queue settings / edit Check snapshot qs 1`] = `
<div
  class="wrap-queues-setting"
>
  <div
    class="wrap-head"
  >
    <h2>
      QUEUES SETTINGS
    </h2>
     
    <svg
      class="arrow down"
      fill="transparent"
      viewBox="0 0 10 18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 1L1 9L9 17"
        stroke="var(--light-blue)"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="table-container"
    >
      <div
        class="available"
      >
        <div
          class="table-title"
        >
          Available Queues
        </div>
         
        <div
          class="table"
        >
          <ul
            class="table-header"
          >
            <li>
              Name of Queue
            </li>
             
            <li>
              Queue Number
            </li>
             
            <li>
              Agents in Queue
            </li>
             
            <li>
              Action
            </li>
          </ul>
           
          <div
            class="table-content"
          />
           
          <div
            class="btn-table-container"
          >
            <button
              class="add"
              disabled="disabled"
              style="display: none;"
            >
              
              Select All
              
              <img
                alt="icon-to"
                src="@/assets/img/icons/table-icon-to.svg"
                width="23"
              />
            </button>
          </div>
        </div>
      </div>
       
      <div
        class="switcher"
      >
        <svg
          fill="none"
          height="66"
          viewBox="0 0 66 66"
          width="66"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M65 17.0005C65 14.7914 63.2092 13.0006 61.0001 13.0006H15.1797L20.5496 7.82879C22.1693 6.26484 22.1693 3.73691 20.5496 2.17296C18.9257 0.609012 16.2968 0.609012 14.6771 2.17296L2.21789 14.1726C0.594037 15.7365 0.594037 18.2645 2.21789 19.8284L14.6771 31.828C15.487 32.608 16.5502 33 17.6134 33C18.6765 33 19.7397 32.608 20.5496 31.828C22.1693 30.2641 22.1693 27.7362 20.5496 26.1722L15.1797 21.0004H61.0001C63.2092 21.0004 65 19.2096 65 17.0005Z"
            fill="white"
            fill-opacity="0.1"
          />
           
          <path
            d="M51.3229 34.173C49.699 32.609 47.0743 32.609 45.4504 34.173C43.8266 35.7369 43.8266 38.2648 45.4504 39.8288L50.8203 45.0006H4.99988C2.79081 45.0006 1 46.7914 1 49.0005C1 51.2096 2.79081 53.0004 4.99988 53.0004H50.8203L45.4504 58.1722C43.8266 59.7362 43.8266 62.2641 45.4504 63.828C46.2603 64.608 47.3234 65 48.3866 65C49.4498 65 50.513 64.608 51.3229 63.828L63.7821 51.8284C65.406 50.2645 65.406 47.7365 63.7821 46.1726L51.3229 34.173Z"
            fill="white"
            fill-opacity="0.1"
          />
           
          <path
            class="stroke-arrow"
            d="M65 17.0005C65 14.7914 63.2092 13.0006 61.0001 13.0006H15.1797L20.5496 7.82879C22.1693 6.26484 22.1693 3.73691 20.5496 2.17296C18.9257 0.609012 16.2968 0.609012 14.6771 2.17296L2.21789 14.1726C0.594037 15.7365 0.594037 18.2645 2.21789 19.8284L14.6771 31.828C15.487 32.608 16.5502 33 17.6134 33C18.6765 33 19.7397 32.608 20.5496 31.828C22.1693 30.2641 22.1693 27.7362 20.5496 26.1722L15.1797 21.0004H61.0001C63.2092 21.0004 65 19.2096 65 17.0005Z"
            stroke-width="2"
          />
           
          <path
            class="stroke-arrow"
            d="M51.3229 34.173C49.699 32.609 47.0743 32.609 45.4504 34.173C43.8266 35.7369 43.8266 38.2648 45.4504 39.8288L50.8203 45.0006H4.99988C2.79081 45.0006 1 46.7914 1 49.0005C1 51.2096 2.79081 53.0004 4.99988 53.0004H50.8203L45.4504 58.1722C43.8266 59.7362 43.8266 62.2641 45.4504 63.828C46.2603 64.608 47.3234 65 48.3866 65C49.4498 65 50.513 64.608 51.3229 63.828L63.7821 51.8284C65.406 50.2645 65.406 47.7365 63.7821 46.1726L51.3229 34.173Z"
            stroke-width="2"
          />
        </svg>
      </div>
       
      <div
        class="selected"
      >
        <div>
          <div
            class="table-title"
          >
            Selected Queues
          </div>
           
          <div
            class="table"
          >
            <ul
              class="table-header"
            >
              <li>
                Name of Queue
              </li>
               
              <li>
                Queue Number
              </li>
               
              <li>
                Agents in Queue
              </li>
               
              <li>
                Action
              </li>
            </ul>
             
            <div
              class="table-content"
            />
             
            <div
              class="btn-table-container"
            >
              <button
                class="remove"
                style="display: none;"
              >
                
                Remove All
                
                <img
                  alt="cross"
                  src="@/assets/img/icons/cross.svg"
                  width="23"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
